_test/test_sqw/split_to_messages_for_testing.m

```diff
@@ -1,35 +1,107 @@
-function [messages,npix1,npix2] = split_to_messages_for_testing(test_pix_block,nbin_start,nbin_end,n_files,fileind)
-% split pix block into messages block
+function [messages,npix1,npix2,last_bin_completed] = split_to_messages_for_testing(test_pix_block,...
+    nbin_start,nbin_end,n_files,fileind,buf_size,npix1)
+% split pix block into messages block. e.g prepare n_files messages, whith
+% pixels, contributing into these bins.
+%
+% inputs:
+% test_pix_block -- array with test information for splitting pixels (see
+%                   build_pix_block_for_testing for the details)
+% nbin_start     -- first bin to prepare for messages
+% nbin_end       -- last bin to prepaste for messages
+% n_files        -- number of messages to split pix_block into (the info
+%                   about splitting mess position is stored in the second
+%                   column of the pix_block)
+% fileind        -- auxiliary information, specifying the indexes of files
+%                   to process messages (numel(fileind)== n_files)
+% Optional:
+% buf_size       -- if present, constrain the number of pixels, moved to
+%                   messages by this number
+% npix1          -- if present, the first pixel to use to define the range
+%                   instead of the first bin number
+%
+% Output:
+% messages       -- cellarray of messages containing pix information. See
+%                   pix_cache about the format of these messages
+%
+%npix1 - number of the first pixel, contributing into the first bin
+%        requested
+%npix2 - number of the last pixel, contributing into the bins requested
+%
+% last_bin_completed -- if true, the messages contain full bins. If false,
+% some pixels from the bin are not fitting the buffer and will be placed in
+% messages later.
+
 if ~exist('fileind','var')
     fileind = 1:n_files;
 end
+if nargin<6
+    npix1 = find(test_pix_block(1,:)>=nbin_start,1);
+end

```

Comment:
Looks as if npix1 is arg#7 if present. Guessing it is possible for filebuf to be present as an argument but not npix1, so maybe this test should be <7.

Reply:
sorry, buf_size

Reply:
the comment is to a previous version of the code. The new one does not have this part